### PR TITLE
(Enhancement) Ginkgo test framework for k8tls

### PR DESCRIPTION
**Pull request by PrimalPimmy**
Added test framework using ginkgo, current has two cases but as more features are added, more tests can be put here.

### CODE REVIEW

**Review by nyrahul — CHANGES_REQUESTED** [review]
Looks mostly good to me. Please check the comments. Thanks

File: .github/workflows/ci-test-ginkgo.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          go-version: "v1.20"
+      
+      - name: Create k3s cluster
+        run: ./.github/workflows/scripts/install_k3s.sh
+
+      - name: Test connectivity
+        run: kubectl get pods -A
+
+      - name: Deploy wordpress-mysql
+        run: kubectl apply -f ./tests/res/wordpress-mysql-dep.yaml
```
Comment (nyrahul) [inline .github/workflows/ci-test-ginkgo.yml:38]: Need to add wait to check if the pods are up and running.

File: .github/workflows/scripts/install_k3s.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+while [[ $(date -u +%s) -le $endtime ]]
+do
+    status=$(kubectl get pods -A -o jsonpath={.items[*].status.phase})
+    [[ $(echo $status | grep -v Running | wc -l) -eq 0 ]] && break
+    echo "wait for initialization"
+    sleep 1
+done
+
+kubectl get pods -A
```
Comment (nyrahul) [inline .github/workflows/scripts/install_k3s.sh:37]:
```suggestion
curl -sfL https://get.k3s.io | K3S_KUBECONFIG_MODE="644" INSTALL_K3S_EXEC="--disable=traefik" sh -
[[ $? != 0 ]] && echo "Failed to install k3s" && exit 1

mkdir -p ~/.kube && cp /etc/rancher/k3s/k3s.yaml ~/.kube/config

kubectl wait --for=condition=ready --timeout=15m -n kube-system pod -l k8s-app=metrics-server
```

File: tests/k8tls-test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
+package k8tlstest
+
+import (
+	"encoding/csv"
+	"os"
+
+	// . "github.com/kubearmor/KubeArmor/tests/util"
+
```
Comment (nyrahul) [inline tests/k8tls-test.go:8]:
```suggestion
```

File: tests/k8tls-test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func matchCSV(file1 string, file2 string) {
```
Comment (nyrahul) [inline tests/k8tls-test.go:13]: This function is clumsy and huge scope for improvisation. Secondly, it is not a generic matchCSV operation. There are fields `address` hardcoded in this API.

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: tests/k8tls-test.go (hunk 4 above)
Comment (PrimalPimmy) [reply tests/k8tls-test.go:13]: What do you mean by clumsy? Also yes I was wondering how I would test something like Address which keeps on changing.

Was thinking of using regex based matching in the Address column, matching address:port formats . No other ideas yet, thoughts?

**Review by nyrahul — APPROVED** [review]

**Pull request merged**